src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -1727,8 +1736,12 @@ namespace EvaporativeCoolers {
             //***************************************************************************
             //                  ENERGY CONSUMED BY THE RECIRCULATING PUMP
             // Add the pump energy to the total Evap Cooler energy comsumption
-            EvapCond(EvapCoolNum).EvapCoolerPower += EvapCond(EvapCoolNum).RecircPumpPower;
-
+            Real64 FlowFraction = 1.0;
+            Real64 MassFlowRateMax = Node(EvapCond(EvapCoolNum).InletNode).MassFlowRateMax;
+            if (MassFlowRateMax > 0) {
+                FlowFraction = EvapCond(EvapCoolNum).InletMassFlowRate / MassFlowRateMax;
```

Comment:
@Nigusse @Myoldmopar @rraustad So, I was puzzled by the meter diffs in 1ZoneEvapCooler and 1ZoneParameterAspect, because these have airloop evap coolers, no zoneHVAC. 

It turns out that 1ZoneParameterAspect is hard-sized with the airloop, fan, and OA controller all at 3.0 but the terminal unit is at 2.5. So, the evap cooler power is always 2.5/3.0 on this branch compared to develop. (1ZoneEvapCooler is similar with 2.0 vs 1.5). I don't see anything in the comments to indicate that the different flow rates are intentional. So, one option here is to simply fix those to use the terminal unit flow rate throughout and then the diffs would go away. 

Since there is no design flow rate input for EvaporativeCooler:*:CelDekPad, using the node max flow rate is a clever solution to calculate a "plr" for the evap cooler, but will this always produce the desired evap cooler power consumption? What if the evap cooler is in a VAV system, does that still make sense?

Thoughts?

Reply:
1) I would expect MassFlowRateMax = branch max flow so this max looks accurate.
2) Does EvapCooler have a VS pump? Probably no since the misters need high pressure to mist water.
3) if EvapCooler cycles then this seems reasonable for pump power, if EvapCooler runs entire time step then previous pump power calc was accurate.
4) Don't care if it's VAV. How long does the pump run this time step is more important.

Reply:
So, perhaps instead of using flow rate as a proxy for PLR, the PLR should default to 1.0 and ZoneHVAC:EvapCooler needs to pass a PLR to the evap cooler component?

Reply:
The outlet conditions are correct now so I was thinking more like:

If (constant fan){
 same power equation as before
} else {
 new PLR equation
}

Reply:
The base evap cooler component doesn't know anything about fan type - only the ZoneHVAC:EvapCooler knows that. If the eval cooler component is loose on a branch, then it doesn't know what the fan will do.

Reply:
@Nigusse you need to figure out a way to know when the fan is cycling vs constant so you can get the correct pump power.

Reply:
If evap cooler is loose on the air loop branch you can use this bool and probably also if AirLoopNum > 0:
AirLoopControlInfo(AirLoopNum).AnyContFan, or could pass PLR = 1 as @mjwitte suggested.

If evap cooler is in the ZoneHVAC:EvapCooler then you need to pass info to the evap cooler to tell it if it is cycling or constant.

Passing PLR as an argument is probably easiest unless you can think of something else.

Reply:
@mjwitte @rraustad I am thinking to pass the ZoneHVAC:EvapCooler unit PLR as an optional argument to SimEvapCooler() function and cascade it to CalcDirectEvapCooler() and CalcDryIndirectEvapCooler() functions. If the  PLR argument is not present, then I will reset it to 1.0.  The later will be used for air loop.

Reply:
@rraustad @mjwitte @Nigusse  Hi, sorry for jumping into the conversation which I might haven't understood thoroughly. So, is there a way to model cycling fan and cycling evap cooler pump in loose air loop? So far I didn't find any approach to directly put cycling fan into air loop and make it cycling properly. And as far as I've learned in this conversation, there's no way to communicate PLR between fan and evap cooler (this PR is going to pass 1.0 for air loop), correct?

Reply:
There's no way presently to model cycling fans on built-up branches. EvaporativeCooler:*:ResearchSpecial is able to control itself to a setpoint.

Reply:
That's true for purely component built-up branches, but when evap cooler is used with parent objects, the fan operating mode schedule is known as AirLoopControlInfo(AirLoopNum).CycFanSchedPtr (as long as this variable is set correctly everywhere) and could, in the future, be used to cycle the evap cooler.